src/containers/views/DataFileEdit.js

```diff
@@ -95,11 +118,16 @@ export class DataFileEdit extends Component {
         className="single">
         {errors.length > 0 && <Errors errors={errors} />}
         <div className="content-header">
-          <Breadcrumbs splat={filename} type="datafiles" />
+          <Breadcrumbs splat={directory || ""} type="data files" />
         </div>
 
         <div className="content-wrapper">
           <div className="content-body">
+            <InputPath
+              onChange={onDataFileChanged}
+              type="data files"
+              path={path} // TODO: Support using `relative_path` from API instead
```

Comment:
You just prepend `_data` to `path` which is the concatenation of the current directory and the relative path that comes from the input in `putDatafile` action. [Here](https://github.com/jekyll/jekyll-admin/blob/master/src/actions/collections.js#L109) is how I managed to do that for collections that won't drop into `pages`.